tests/backup_helper.go

```
@@ -1006,6 +1006,115 @@ func CreateBackupWithCustomResourceTypeWithValidation(ctx context1.Context, back
+				go func(i int) {
+					defer innerWg.Done()
+					defer GinkgoRecover()
+
+					select {
+					case semaphore <- struct{}{}:
+						defer func() { <-semaphore }()
+					case <-ctx.Done():
+						return
+					}
+					currentBackupName := fmt.Sprintf("%s-%s-%d", backupNamePrefix, RandomString(8), i+1)
+					err := CreateBackup(currentBackupName, clusterName, backupLocationName, backupLocationUid, []string{scheduledAppContext.ScheduleOptions.Namespace}, labelSelectors, backupOrgID, currentClusterUid, "", "", "", "", ctx)
+					backupResults <- backupResult{name: currentBackupName, err: err, bkpContext: scheduledAppContext}
+				}(i)
+			}
+
+			innerWg.Wait()
+		}(scheduledAppContext)
+	}
+	go func() {
+		wg.Wait()
+		close(backupResults)
+	}()
+
+	backupMap := make(map[string]*scheduler.Context)
+	for result := range backupResults {
+		if result.err != nil {
+			log.Errorf("Failed to create backup: %v", result.err)
+			errList = append(errList, result.err)
+			cancel()
+			break
+		} else {
+			log.Infof("Successfully created backup [%s]", result.name)
+			backupNameList = append(backupNameList, result.name)
+			backupMap[result.name] = result.bkpContext
+		}
+	}
+
+	if len(errList) > 0 {
+		return backupNameList, fmt.Errorf("some backups failed")
+	}
+	validationResults := make(chan error)
+
+	// Validate all the backups.
+	for backupName, bkpContext := range backupMap {
+		wg.Add(1)
+		go func(backupName string, bkpContext *scheduler.Context) {
+			defer wg.Done()
+			defer GinkgoRecover()
+
+			err := ValidateBackup(ctx, backupName, backupOrgID, []*scheduler.Context{bkpContext}, nil)
+			validationResults <- err
+		}(backupName, bkpContext)
+	}
+	go func() {
+		wg.Wait()
+		close(validationResults)
+	}()
+	for err := range validationResults {
+		if err != nil {
+			log.Errorf("Validation failed for backup: %v", err)
+			errList = append(errList, err)
+		}
+	}
+	if len(errList) > 0 {
+		return backupNameList, fmt.Errorf("some validations failed")
```

Lets return the error as well in the string.